libs/wingc/src/capture.rs

```diff
@@ -312,22 +285,33 @@ fn scan_captures_in_expression(
 					}
 				}
 			}
-			Reference::NestedIdentifier { object, property: _ } => {
+			Reference::NestedIdentifier { object, property } => {
 				res.extend(scan_captures_in_expression(object, env, statement_idx, diagnostics));
 
-				// If the expression evaluates to a resource we should check if we need to capture the property as well
-				// TODO: do we really need this? I think we capture `object` above recursively and therefore don't need special handling of `property`.
-				// if let &Type::ResourceObject(resource) = object.evaluated_type.borrow().unwrap().into() {
-				// 	let resource = resource.as_resource().unwrap();
-				// 	let (prop_type, flight) = resource.env.lookup_ext(property);
-				// 	if prop_type.as_resource_object().is_some() && matches!(flight, Flight::Pre) {
-				// 		debug!(
-				// 			"We seem to be accessing the preflight resource {} {} inflight!",
-				// 			r, property.span
-				// 		);
-				// 		res.push(r);
-				// 	}
-				// }
+				// If the expression evaluates to a resource we should check what method of the resource we're accessing
+				if let Type::Resource(ref resource) = **object.evaluated_type.borrow().as_ref().unwrap() {
+					let (prop_type, _flight) = match resource.env.lookup_ext(property, None) {
+						Ok((prop_type, phase)) => (
+							prop_type
+								.as_variable()
+								.expect("Expected resource property to be a variable")
+								._type,
+							phase,
+						),
+						Err(type_error) => {
+							panic!("{}", type_error);
+						}
+					};
+
+					let func = prop_type.as_function_sig().unwrap();
```

Comment:
We're not in _call_ anymore, this is the generic scan captures function for _expressions_ so I don't think we can simply `unwrap` here and assume it'll be a function sig.
On the other hand I'm not sure why we even have all this "if the expression evaluates to a resource we should check what method of the resource we're accessing". This doesn't affect the otuput all it does is adds a debug print. Maybe worth keeping for the future, but you'll need to fix this to verify `prop_type` is a function sig.

Reply:
Good catch, added logic to handle that. This was indeed failing in the case of something like `this._counter.add()`. Also made a comment noting that if we want to add some sort of capturing logic inside a reference while recursing it, this is probably where it'd go.